### ISSUE TITLE
Object Detection does not work on DIGITS 6 Docker

### ISSUE THREAD

**Issue by mrteera** [NVIDIA/DIGITS#1800]
I follow this tutorial https://github.com/NVIDIA/DIGITS/tree/master/examples/object-detection on DIGITS 6 Docker. My system is GTX960m and CUDA version 8.0.61 (inside Docker).

I have got this error when I try to train a model:
```
Check failed: error == cudaSuccess (8 vs. 0)  invalid device function
```

**Comment by juiceboxjoe** [comment]
@mrteera Have you found a solution to this problem? I'm having the exact same error. My system is the same as yours:

- GTX960m
- CUDA version 8.0.61 
- caffe version 0.15.14

I'm trying to use the DIGITS 6 Docker image using nvidia-docker.

My goal is to fine-tune some existing weights but when I start the fine-tuning process I get the exact same error as @mrteera:

> val_data does not need backward computation.
This network produces output loss_bbox
This network produces output loss_coverage
This network produces output mAP
This network produces output precision
This network produces output recall
Network initialization done.
Solver scaffolding done.
Finetuning from /jobs/20170915-002921-3323/model.caffemodel
Ignoring source layer train_data
Ignoring source layer train_label
Ignoring source layer train_transform
Starting Optimization
Solving
Learning Rate Policy: step
Iteration 0, Testing net (#0)
Ignoring source layer train_data
Ignoring source layer train_label
Ignoring source layer train_transform
Check failed: error == cudaSuccess (8 vs. 0)  invalid device function
      
 `caffe device_query -gpu 0` returns:
> libdc1394 error: Failed to initialize libdc1394
I0918 16:25:09.779546   183 caffe.cpp:118] Querying GPUs 0
I0918 16:25:09.785096   183 common.cpp:193] Device id:                     0
I0918 16:25:09.785114   183 common.cpp:194] Major revision number:         5
I0918 16:25:09.785118   183 common.cpp:195] Minor revision number:         0
I0918 16:25:09.785125   183 common.cpp:196] Name:                          GeForce GTX 960M
I0918 16:25:09.785130   183 common.cpp:197] Total global memory:           2100232192
I0918 16:25:09.785137   183 common.cpp:198] Total shared memory per block: 49152
I0918 16:25:09.785142   183 common.cpp:199] Total registers per block:     65536
I0918 16:25:09.785148   183 common.cpp:200] Warp size:                     32
I0918 16:25:09.785153   183 common.cpp:201] Maximum memory pitch:          2147483647
I0918 16:25:09.785159   183 common.cpp:202] Maximum threads per block:     1024
I0918 16:25:09.785164   183 common.cpp:203] Maximum dimension of block:    1024, 1024, 64
I0918 16:25:09.785171   183 common.cpp:206] Maximum dimension of grid:     2147483647, 65535, 65535
I0918 16:25:09.785176   183 common.cpp:209] Clock rate:                    1176000
I0918 16:25:09.785181   183 common.cpp:210] Total constant memory:         65536
I0918 16:25:09.785187   183 common.cpp:211] Texture alignment:             512
I0918 16:25:09.785192   183 common.cpp:212] Concurrent copy and execution: Yes
I0918 16:25:09.785197   183 common.cpp:214] Number of multiprocessors:     5
I0918 16:25:09.785202   183 common.cpp:215] Kernel execution timeout:      Yes

`dpkg -l | grep 'cudart\|libcudnn\|libnccl\|caffe\|torch\|digits'` returns:
>  ii  cuda-cudart-8-0                    8.0.61-1                                   amd64        CUDA Runtime native Libraries
ii  libcudnn5                          5.1.10-1+cuda8.0                           amd64        cuDNN runtime libraries
ii  libnccl1                           1.2.3-1+cuda8.0                            amd64        NVIDIA Collectives Communication Library (NCCL) Runtime
ii  torch7-nv                          0.9.99-1+cuda8.0                           amd64        NVidia Torch Bundle (with CUDA). Made for DIGITS.


Everything is fine on the DIGITS 5 docker. According to #55 cuDNN support is not included in DIGITS Docker images. I'm trying to use the DIGITS 6 Docker image since, judging by the Dockerfile, cuDNN support is actually included in it. Any help would be appreciated.

**Comment by mrteera** [comment]
@juiceboxjoe Currently, I temporary stop using Digits.. I just use normal Tensorflow instead, probably until Digits 6 final release.

**Comment by usrec** [comment]
Same error... same card... GTX960m

**Comment by ethantang95** [comment]
Was an appropriate NVidia driver installed upon that? Are you using this in an Ubuntu or Windows environment?

**Comment by usrec** [comment]
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 384.90                 Driver Version: 384.90                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 960M    Off  | 00000000:01:00.0 Off |                  N/A |
| N/A   43C    P8    N/A /  N/A |    352MiB /  2002MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1199      G   /usr/lib/xorg/Xorg                           198MiB |
|    0      2211      G   /usr/bin/compiz                              114MiB |
|    0      2889      G   /usr/lib/firefox/firefox                       1MiB |
|    0     14004      C   /usr/lib/libreoffice/program/soffice.bin      25MiB |
+-----------------------------------------------------------------------------+
Using Ubuntu

**Comment by ethantang95** [comment]
Do any other network or examples work?

**Comment by usrec** [comment]
Tried googlenet and alexnet.  Both failed.  My dataset(stanford dogs) appeared to load correctly.  I didn't try a second data set.

**Comment by ethantang95** [comment]
@usrec are you running this inside docker or on the base OS?

**Comment by usrec** [comment]
nvidia-docker:
sudo nvidia-docker run --name digits -d -p 5000:5000 -v /home/malibucool/dogs/dogs:/dogs/dogs nvidia/digits

**Comment by ethantang95** [comment]
Which version of nvidia docker are you using?

**Comment by usrec** [comment]
NVIDIA Docker: 1.0.1

Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:42:18 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:56 2017
 OS/Arch:      linux/amd64
 Experimental: false
